### PR TITLE
feat: update JavaScript client library spec for MFA (Phone)

### PR DESCRIPTION
**Pull request by J0**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.
YES

## What kind of change does this PR introduce?

Adds the library specification for MFA (Phone)  JavaScript client library bindings. 

TODO:
- [x] update Python, Swift, and Flutter lib maintainers once done

### CODE REVIEW

File: apps/docs/spec/supabase_js_v2.yml [hunk 1]
```diff
@@ -2281,12 +2282,39 @@ functions:
...
+        code: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.enroll({
+            factorType: 'phone',
+            friendlyName: 'your_friendly_name',
+            phone_number: '+12345678',
+          })
+
+          // Use the id to create a challenge and send an SMS with a code to the user.
+          const { id, type, friendly_name, phone } = data
```
Comment (hf) [inline apps/docs/spec/supabase_js_v2.yml:2297]: Show the challenge code below.

File: apps/docs/spec/supabase_js_v2.yml [hunk 2]
```diff
@@ -2281,12 +2282,39 @@ functions:
             error: null
           }
           ```
+      - id: enroll-phone-factor
+        name: Enroll a Phone Factor
```
Comment (hf) [inline apps/docs/spec/supabase_js_v2.yml:2286]: In general for these examples, let's do two:

- One for enrolling (add new factor)
- One for verifying (after sign-in)

File: apps/docs/spec/supabase_js_v2.yml [hunk 3]
```diff
@@ -2281,12 +2283,41 @@ functions:
...
+        isSpotlight: false
+        code: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.enroll({
+            factorType: 'phone',
+            friendlyName: 'your_friendly_name',
+            phone_number: '+12345678',
```
Comment (J0) [inline apps/docs/spec/supabase_js_v2.yml:2294]:
```suggestion
            phone: '+12345678',
```

File: apps/docs/spec/supabase_js_v2.yml [hunk 4]
```diff
@@ -2307,6 +2338,45 @@ functions:
...
+          ```js
+          const { data, error } = await supabase.auth.mfa.challenge({
+            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225',
+          })
+          ```
+        response: |
+          ```json
+          {
+            data: {
+              id: '<ID>',
```
Comment (J0) [inline apps/docs/spec/supabase_js_v2.yml:2354]:
```suggestion
              id: '<ID>',
              type: 'phone',
```

File: apps/docs/spec/supabase_js_v2.yml [hunk 5]
```diff
@@ -2307,6 +2338,45 @@ functions:
...
+          const { data, error } = await supabase.auth.mfa.challenge({
+            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225',
+            channel: 'whatsapp',
+          })
+          ```
+        response: |
+          ```json
+          {
+            data: {
+              id: '<ID>',
```
Comment (J0) [inline apps/docs/spec/supabase_js_v2.yml:2374]:
```suggestion
              id: '<ID>',
              type: 'phone',
```

File: apps/docs/spec/supabase_js_v2.yml [hunk 6]
```diff
@@ -2234,21 +2234,22 @@ functions:
...
+      - Use `totp` or `phone` as the `factorType` and use the returned `id` to create a challenge.
       - To create a challenge, see [`mfa.challenge()`](/docs/reference/javascript/auth-mfa-challenge).
       - To verify a challenge, see [`mfa.verify()`](/docs/reference/javascript/auth-mfa-verify).
-      - To create and verify a challenge in a single step, see [`mfa.challengeAndVerify()`](/docs/reference/javascript/auth-mfa-challengeandverify).
+      - To create and verify a TOTP challenge in a single step, see [`mfa.challengeAndVerify()`](/docs/reference/javascript/auth-mfa-challengeandverify).
       - To generate a QR code for the `totp` secret in nextjs, you can do the following:
       ```html
       <Image src={data.totp.qr_code} alt={data.totp.uri} layout="fill"></Image>
       ```
+      - The `challenge` and `verify` steps are separated when using Phone factors as the user will need time to receive and input the code obtanied from the SMS in challenge.
```
Comment (J0) [inline apps/docs/spec/supabase_js_v2.yml:2252]:
```suggestion
      - The `challenge` and `verify` steps are separated when using Phone factors as the user will need time to receive and input the code obtained from the SMS in challenge.
```

File: apps/docs/spec/supabase_js_v2.yml [hunk 7]
```diff
@@ -2234,21 +2234,22 @@ functions:
...
     title: 'mfa.enroll()'
     $ref: '@supabase/auth-js.GoTrueMFAApi.enroll'
     notes: |
-      - Currently, `totp` is the only supported `factorType`. The returned `id` should be used to create a challenge.
+      - Use `totp` or `phone` as the `factorType` and use the returned `id` to create a challenge.
       - To create a challenge, see [`mfa.challenge()`](/docs/reference/javascript/auth-mfa-challenge).
       - To verify a challenge, see [`mfa.verify()`](/docs/reference/javascript/auth-mfa-verify).
-      - To create and verify a challenge in a single step, see [`mfa.challengeAndVerify()`](/docs/reference/javascript/auth-mfa-challengeandverify).
+      - To create and verify a TOTP challenge in a single step, see [`mfa.challengeAndVerify()`](/docs/reference/javascript/auth-mfa-challengeandverify).
       - To generate a QR code for the `totp` secret in nextjs, you can do the following:
```
Comment (hf) [inline apps/docs/spec/supabase_js_v2.yml:2248]:
```suggestion
      - To generate a QR code for the `totp` secret in Next.js, you can do the following:
```

**Review by hf — APPROVED** [review]
Looks good!

**Comment by J0** [comment]
Going to hold off on taking further action on this PR until docs changes and UI changes are in

Also need to check if we should add more emphasis that it is a Pro feature and available only with Add-on purchase

**Comment by hf** [comment]
Also please update the PR description to use human text. :D

**Pull request merged**